### ISSUE TITLE
provider states are not optional, they should be, your doc says they are

### ISSUE THREAD

**Issue by thetrav** [pact-foundation/pact-js#54]
I have a stateless service (provider) I'm trying to test.

There is one state "server running" which requires no setup.  
When attempting to test using pact-js, I get an assertion error:
```
1.1) Failure/Error: set_up_provider_state interaction.provider_state, options[:consumer]

          RuntimeError:
            Could not find provider state "server is running" for consumer ReferenceClient

     1.2) Failure/Error: tear_down_provider_state interaction.provider_state, options[:consumer]

          RuntimeError:
            Could not find provider state "server is running" for consumer ReferenceClient
```

Additionally I'm getting the ruby help messaging back for setting up provider state, which is obviously no help for javascript based testing

**Comment by thetrav** [comment]
So, it appears the state key is not required in the consumer spec.  
Removing that allows me to work around this issue.

**Comment by mefellows** [comment]
Thanks, yes the Ruby messaging is not very helpful (we use it under the hood). We are working on a native implementation but perhaps there are some tactical options we can implement to make this easier.

As for the main problem, if you use provider states in your tests - in your case "server running" - then you must provide `providerStatesSetupUrl` and `providerStatesUrl`. Perhaps I can make this clearer.

As you noted, if the provider state is not actually a 'state' then omitting it makes more sense.

I'm going to make this for enhancement.

**Comment by mefellows** [comment]
Also I think @bethesque has made some changes to the verifier, once we get this merged it will hopefully simplify this a little bit (removing one of these APIs).

**Comment by mefellows** [comment]
This was fixed in the last release `v2.6.0`, and docs have been updated to indicate when it is required. Closing.
